sw/device/rom_exts/rom_ext_manifest_parser.h

```diff
@@ -0,0 +1,213 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_ROM_EXTS_ROM_EXT_MANIFEST_PARSER_H_
+#define OPENTITAN_SW_DEVICE_ROM_EXTS_ROM_EXT_MANIFEST_PARSER_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/rom_exts/manifest.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * ROM Extension image signature.
+ */
+typedef struct rom_ext_signature {
+  uint32_t data[ROM_EXT_IMAGE_SIGNATURE_SIZE_WORDS];
+} rom_ext_signature_t;
+
+/**
+ * ROM Extension lockdown information.
+ *
+ * TODO - probably would eventually become an internal type, and there will
+ *        be another public type with more finely parsed information.
+ */
+typedef struct rom_ext_lockdown_info {
+  uint32_t data[ROM_EXT_PERIPHERAL_LOCKDOWN_INFO_SIZE_WORDS];
+} rom_ext_lockdown_info_t;
+
+/**
+ * ROM Extension public key.
+ */
+typedef struct rom_ext_public_key {
+  uint32_t data[ROM_EXT_SIGNATURE_PUBLIC_KEY_SIZE_WORDS];
+} rom_ext_public_key_t;
+
+/**
+ * ROM Extension image extension.
+ */
+typedef struct rom_ext_extension {
+  uint32_t offset;
+  uint32_t checksum;
+} rom_ext_extension_t;
+
+/**
+ * Retrieves the ROM_EXT identifier.
+ *
+ * The memory address where ROM_EXT identifier field resides, is relative to
+ * `base_addr`.
+ *
+ * @param base_addr Base address of the ROM_EXT manifest.
+ * @return ROM_EXT identifier.
+ */
+uint32_t rom_ext_identifier_get(mmio_region_t base_addr);
```

Comment:
let's wrap these `base_addr` in a struct like:
```c
typedef struct rom_ext_manifest {
  mmio_region_t base_addr;
} rom_ext_manifest_t
```

and then all the functions work on a `rom_ext_manifest_t`, which we can create in unit tests.

We should also add a function that calculates the `rom_ext_manifest_t` based on slot A vs slot B, which wouldn't be used in mocked unit tests, but should be tested.

Reply:
Is there a particular reason we're using mmio_region? AIUI the ROM_EXT lives in flash, which can be read via normal pointers. It feels like it would be more natural to work in terms of a `const uint32_t*`, seeing as that would vastly simplify the tests.

Reply:
> We should also add a function that calculates the rom_ext_manifest_t based on slot A vs slot B, which wouldn't be used in mocked unit tests, but should be tested.

Had it, removed it, will add it back.

Reply:
> s there a particular reason we're using mmio_region? AIUI the ROM_EXT lives in flash, which can be read via normal pointers. It feels like it would be more natural to work in terms of a const uint32_t*, seeing as that would vastly simplify the tests.

For simplicity, subject to change later, but doesn't affect the public interface when wrapped in `rom_ext_manifest_t`.

Reply:
Done.